src/main/java/iterator/RichIterator.java

```diff
@@ -452,7 +795,53 @@ default RichIterator<A> dropUntil(Predicate<? super A> predicate) {
      * @return the iterator
      */
     default BufferedIterator<A> buffered() {
-        throw new NotImplementedException();
+        return new BufferedIterator<A>() {
+            private A head;
+
+            @Override
+            public boolean hasNext() {
+                return headOptional().isPresent();
+            }
+
+            @Override
+            public A next() {
+                Optional<A> element = headOptional();
+                Optional<A> newHead = RichIterator.this.nextOptional();
```

Comment:
This will read the next element although it is unnecessary in case we already have a head. (I should add test for that).